pkg/wireguard/wireguard.go

```diff
@@ -552,6 +552,11 @@ func PrepareDeviceConfig(peerEvents []PeerEvent, oldCfg *wgtypes.Device, userHan
 
 func checkDuplicateUpdates(seq iter.Seq[PeerEvent], oldCfg *wgtypes.Device, logger *zap.Logger) iter.Seq[PeerEvent] {
 	return iter.Filter(seq, func(peerEvent PeerEvent) bool {
+		// never skip remove
```

Comment:
what if that's add and remove later on?

my point we should dedup in some more intelligent way?

Reply:
or... in other terms what if the sequence of events is like:

`add` -> `remove` -> `add` ?

Reply:
do you mean `update` -> `update` -> `remove` should turn into a single `remove`?
That's already ok and is covered by the test case.

Reply:
I mean exactly what I posted above, as I think 3rd `add` will get de-duplicated, while `remove` will be executed. 

If we remove, we should only execute it if the peer is there... right?

Reply:
Or is there additional step when we combine same events for the same pubkey?

Reply:
yep, there's. It sorts, then deduplicates

Reply:
so the proper check for 'remove' should be that it doesn't exist in the peers list?

Reply:
sorry, that it actually exists in the peer list

Reply:
I even checked that it works this way by writing the test case. Can commit it I guess.

Reply:
Yeah, good point. Will add that

Reply:
updated, should be good now.